microk8s-resources/actions/common/utils.sh

```diff
@@ -632,25 +632,40 @@ create_user_kubeconfigs() {
   create_kubeconfig_x509 "kubelet.config" "system:node:${hostname}" ${SNAP_DATA}/certs/kubelet.crt ${SNAP_DATA}/certs/kubelet.key ${SNAP_DATA}/certs/ca.crt
 }
 
-create_worker_kubeconfigs() {
+create_worker_kubeconfigs_local_apiserver() {
   hostname=$(hostname | tr '[:upper:]' '[:lower:]')
   create_kubeconfig_x509 "proxy.config" "system:kube-proxy" ${SNAP_DATA}/certs/proxy.crt ${SNAP_DATA}/certs/proxy.key ${SNAP_DATA}/certs/ca.remote.crt
   create_kubeconfig_x509 "kubelet.config" "system:node:${hostname}" ${SNAP_DATA}/certs/kubelet.crt ${SNAP_DATA}/certs/kubelet.key ${SNAP_DATA}/certs/ca.remote.crt
 }
 
+create_worker_kubeconfigs_remote_apiserver() {
+  Create worker configs for an API server that is provided
+  # $1: API server IP
+  # $2: API server port
+  
+  hostname=$(hostname | tr '[:upper:]' '[:lower:]')
+  create_kubeconfig_x509 "proxy.config" "system:kube-proxy" ${SNAP_DATA}/certs/proxy.crt ${SNAP_DATA}/certs/proxy.key ${SNAP_DATA}/certs/ca.remote.crt $1 $2
+  create_kubeconfig_x509 "kubelet.config" "system:node:${hostname}" ${SNAP_DATA}/certs/kubelet.crt ${SNAP_DATA}/certs/kubelet.key ${SNAP_DATA}/certs/ca.remote.crt $1 $2
+}
+
 create_kubeconfig_x509() {
   # Create a kubeconfig file with x509 auth
   # $1: the name of the config file
   # $2: the user to use al login
   # $3: path to certificate file
   # $4: path to certificate key file
   # $5: path to ca file
+  # $6: (optional) API server IP, optional defaults to 127.0.0.1
+  # $7: (optional) API server port, optional defaults to the port of the current node
 
   kubeconfig=$1
   user=$2
   cert=$3
   key=$4
   ca=$5
+  apiserver=${6:-"127.0.0.1"}
+  apiserver_port="$(cat $SNAP_DATA/args/kube-apiserver | grep -- "--secure-port" | tr "=" " " | gawk '{print $2}')"
```

Comment:
`apiserver_port` is set just below, so the optional value would be ignored

Reply:
Not sure I get this. The default value of `port` is the value of `apiserver_port`. `port` is what is written in the config file.